code/modules/shuttle/supply.dm

```diff
@@ -596,3 +589,135 @@ GLOBAL_LIST_EMPTY(exports_types)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["order"])
+		if(href_list["order"] == "categories")
+			//all_supply_groups
+			//Request what?
+			last_viewed_group = "categories"
+			temp = "<b>Supply points: [round(SSpoints.supply_points)]</b><BR>"
+			temp += "<A href='?src=\ref[src];mainmenu=1'>Main Menu</A><HR><BR><BR>"
+			temp += "<b>Select a category</b><BR><BR>"
+			for(var/supply_group_name in GLOB.all_supply_groups)
+				temp += "<A href='?src=\ref[src];order=[supply_group_name]'>[supply_group_name]</A><BR>"
+		else
+			last_viewed_group = href_list["order"]
+			temp = "<b>Supply points: [round(SSpoints.supply_points)]</b><BR>"
+			temp += "<A href='?src=\ref[src];order=categories'>Back to all categories</A><HR><BR><BR>"
+			temp += "<b>Request from: [last_viewed_group]</b><BR><BR>"
+			for(var/supply_type in SSshuttle.supply_packs)
+				var/datum/supply_packs/N = SSshuttle.supply_packs[supply_type]
+				if(N.hidden || N.contraband || N.group != last_viewed_group) continue								//Have to send the type instead of a reference to
+				temp += "<A href='?src=\ref[src];doorder=[N.name]'>[N.name]</A> Cost: [round(N.cost)]<BR>"		//the obj because it would get caught by the garbage
+
+	if(href_list["send"])
+		if(!linked_beacon)
+			icon_state = "req_tablet_off"
+			playsound(get_turf(src), 'sound/machines/triple_beep.ogg', 25, TRUE)
+			return
+		var/list/turfs_to_land_on = list()
+		for(var/turf/open/floor/T in range(2, linked_beacon))//Deliver around the beacon linked
+			var/area/A = get_area(T)
+			if(A.ceiling != CEILING_GLASS && A.ceiling != CEILING_NONE)
+				return
+			turfs_to_land_on += T
+
+		for(var/datum/supply_order/SO in SSshuttle.shoppinglist)
+			if(!length(turfs_to_land_on))
+				break
+			var/turf/picked_turf = pick(turfs_to_land_on)
+			new /obj/effect/DPtarget(picked_turf, podType, SO.pack)
+			turfs_to_land_on -= picked_turf
+			SSshuttle.shoppinglist -= SO
+
+	if (href_list["clearreq"])
+		SSshuttle.requestlist.Cut()
+		temp = "List cleared.<BR>"
+		temp += "<BR><A href='?src=\ref[src];mainmenu=1'>OK</A>"
+
+	if (href_list["mainmenu"])
+		temp = null
+
+	if (href_list["doorder"])
+		var/datum/supply_packs/P = SSshuttle.supply_packs[href_list["doorder"]]
+		temp = "Invalid Request"
+		temp += "<BR><A href='?src=\ref[src];order=[last_viewed_group]'>Back</A>|<A href='?src=\ref[src];mainmenu=1'>Main Menu</A>"
+		if(SSpoints.supply_points >= round(P.cost))
+			SSpoints.supply_points -= round(P.cost)
+			var/datum/supply_order/O = new /datum/supply_order()
+			O.id = SSshuttle.ordernum
+			O.pack = P
+			SSshuttle.shoppinglist += O
+			P.cost = P.cost * SUPPLY_COST_MULTIPLIER
+			temp = "Thank you for your order.<BR>"
+			temp += "<BR><A href='?src=\ref[src];order=[last_viewed_group]'>Back</A>|<A href='?src=\ref[src];mainmenu=1'>Main Menu</A>"
+		else
+			temp = "Not enough supply points.<BR>"
+			temp += "<BR><A href='?src=\ref[src];order=[last_viewed_group]'>Back</A>|<A href='?src=\ref[src];mainmenu=1'>Main Menu</A>"
+
```

Comment:
Should this generate an order or return?

Reply:
No order should be made